### ISSUE TITLE
Switch to using *Builders in the native client

### ISSUE THREAD

**Issue by michaelklishin** [clojurewerkz/elastisch#95]
ElasticSearch client post 1.0 uses `*Builder` classes for requests. We should use them where possible (we currently instantiate `*Request`s directly in most places).

**Comment by quimrstorres** [comment]
Hey @michaelklishin ,

How do you see this change being implemented? The `clojurewerkz.elastisch.native.conversion` namespace would have a bunch of `->*-builder` functions to replace the `->*-request` ones?

I've spent some time looking into this but the Elasticsearch docs on the site does not help a lot. I try to see how the previous versions of the Java client worked when compared to the current ones, but sometimes the examples correspond to the current version only (even when I switch to the old ones).

**Comment by michaelklishin** [comment]
@quimrstorres yup. The conversion functions are not part of the public API, so don't worry about introducing breaking changes. I wouldn't object if this went into 3.0, too.

**Comment by quimrstorres** [comment]
@michaelklishin cool, thanks!
